cocos/network/CCDownloader-apple.mm

```diff
@@ -396,24 +398,47 @@ - (void)URLSession:(NSURLSession *)session task :(NSURLSessionTask *)task
                              [error.localizedDescription cStringUsingEncoding:NSUTF8StringEncoding],
                              buf);
         }
-        else if(![wrapper get]->storagePath.length()) {
+        else if (![wrapper get]->storagePath.length())
+        {
             // call onTaskFinish for a data task
             // (for a file download task, callback is called in didFinishDownloadingToURL)
             std::string errorString;
-            
+
             const int64_t buflen = [wrapper totalBytesReceived];
             char buf[buflen];
-            
             [wrapper transferDataToBuffer:buf lengthOfBuffer:buflen];
-            
             std::vector<unsigned char> data(buf, buf + buflen);
             
             _outer->onTaskFinish(*[wrapper get],
                                  cocos2d::network::DownloadTask::ERROR_NO_ERROR,
-                                  0,
+                                 0,
                                  errorString,
                                  data);
         }
+        else
+        {
+            NSInteger statusCode = ((NSHTTPURLResponse*)task.response).statusCode;
+            std::vector<unsigned char> buf; // just a placeholder
+            const char *orignalURL = [task.originalRequest.URL.absoluteString cStringUsingEncoding:NSUTF8StringEncoding];
+            std::string errorMessage = "";
+            
+            // Check for error status code
+            if (statusCode >= 400)
+            {
+                errorMessage = cocos2d::StringUtils::format("Downloader: Failed to download %s with status code (%d)", orignalURL, (int)statusCode);
+            }
+            // Unknown issue
+            else
```

Comment:
It's in `URLSession(_:task:didCompleteWithError:):`, so I assume there must be something wrong.